rir/src/compiler/opt/scope_resolution.cpp

```diff
@@ -116,6 +116,9 @@ class TheScopeResolution {
                             !ldfun) {
                             auto hasAllInputs = [&](BB* load) -> bool {
                                 return aval.checkEachSource([&](ValOrig& src) {
+                                    // we cannot move the phi above its src
+                                    if (src.origin->bb() == load)
```

Comment:
Sorry, it is already merged, but shouldn't this check goes inside isPredecessor?

Reply:
I am not sure, in some sense they are predecessors (via a cycle) and this is a special case that we don't want to cross the definition..